contracts/Overtime/SportsAMMV2.sol

```diff
@@ -757,11 +773,39 @@ contract SportsAMMV2 is Initializable, ProxyOwned, ProxyPausable, ProxyReentranc
         );
     }
 
-    function _exerciseTicket(address _ticket) internal {
+    function _exerciseTicket(address _ticket, address _exerciseCollateral, bool _inEth) internal {
         Ticket ticket = Ticket(_ticket);
-        ticket.exercise();
-        IERC20 ticketCollateral = ticket.collateral();
+        (uint userWinningAmount, address ticketOwner, IERC20 ticketCollateral) = ticket.exercise(_exerciseCollateral);
         uint amount = ticketCollateral.balanceOf(address(this));
+        if (userWinningAmount > 0 && _exerciseCollateral != address(0) && _exerciseCollateral != address(ticketCollateral)) {
+            // Off ramp methods
+            // Work in progress:
+            uint offramped;
+            if (_inEth) {
+                // Multi-collateral contract needs changes, for example to swap from ARB -> ETH
+                offramped = multiCollateralOnOffRamp.offrampFromIntoEth(address(ticketCollateral), userWinningAmount);
+                // this method is used as a suggestion from here:
+                // https://solidity-by-example.org/sending-ether/
+                (bool sent, ) = payable(ticketOwner).call{value: offramped}("");
+                require(sent, "Failed to send Ether");
+            } else {
+                // Multi-collateral contract needs changes, for example to swap from ARB -> USDT
+                // Approval code can be deleted for size/gas savings.
+                if (
+                    IERC20(_exerciseCollateral).allowance(address(this), address(multiCollateralOnOffRamp)) <
+                    userWinningAmount
```

Comment:
I think allowance should be checked on `ticketCollateral` not `_exerciseCollateral`.

Reply:
Its always default collateral, and MultiCollateral contract has infinite approval, so not needed.

Reply:
will modify to defaultCollateral